### ISSUE TITLE
A great file app that I am not able to make it work

### ISSUE THREAD

**Issue by killerkaos** [samuelncui/yatm#6]
I want to say it's an excellent idea for the author to attempt an open-source LTFS tool since there is a great need for people like us who don't want to use proprietary software with their tape system. I am having issues installing this app in Ubuntu (22.04.1 LTS). The following are the problems/questions I have during my installation:

1. The curl automatic installation didn't work. I get the following "curl: (23) Failed writing body" error. So, I did a manual installation and was able to get the WebGUI online
2. But it didn't show the tape drive with its content (see below image). I noticed that the drive was not mounted. I did what the author recommended:
   - I found out my "dev/tape/by-id/scsi-" tape drive id by using "ls -al /dev/tape/by-id/"
   - I copied it to the device path to config.yaml, and restarted the service multiple times (and the Ubuntu, too, just in case)
3. The tape drive still didn't show up in the WebGUI. So I am not sure what else to do. Do I need to execute the scripts in script folder (if it is required), or if drivers of LTFS have to be installed in Ubuntu (which I couldn't find the binaries; all of them are for RHE), or something else is going on.

<img width="997" alt="001" src="https://github.com/samuelncui/yatm/assets/27391929/966e4b12-9de5-4950-92a6-aadc386ea26d">

Hardware used (they work great in Windows Server 2022)
- HPE 1/8 G2 Autoloader
- HPE SAS Ultrium 6-SCSI tape drive Model AQ288D#103 C0H27A
- LTO 6 Fujifilm Barium Ferrite tape

Much appreciated for the author for his hard work!

**Comment by killerkaos** [comment]
By the way, it would be great if we could have more documentation on how to initiate the app in Ubuntu.

**Comment by samuelncui** [comment]
1. "curl: (23) Failed writing body" I can't repeat this error, so I'll visit it later.
2. "didn't show the tape drive with its content" This image is the backup job creation page, which only creates backup jobs. After creating a backup job (by dropping files into the waitlist and clicking 'create backup job' button), you can use 'load tape' button on job management page to select a tape drive with empty tape. If there is no tape drive in load tape modal, you can give me a screenshot for the load tape modal and config.yaml.
3. You can use OpenLTFS and compile it yourself. If there is some issues on lib dependency, you can consult those Ubuntu docker image definition at https://github.com/orgs/LinearTapeFileSystem/repositories?type=all.

Autoloader is not tested, but you can add some corresponding commands to those scripts under the `scripts` folder.

**Comment by patrickfitz** [comment]
I have a similar problem- this is not working with IBM HH LT05 drive, the IBM LTFS driver version 2.5.0.0. Installed as root. OS is openSUSE Leap 15.5. 

The drive works from the command line - I can mount it from /dev/sg7 and copy files to it - but installing and following the readme didn't yield a functional system, as follows:

* it doesn't interact with the drive at all
- the default config mentions a different type of path (/dev/tape/by-id/scsi-HUJ0000000) than the recommended "sg" path
- neither work in my case
- the sg_map command gives me the following:  **/dev/sg7  /dev/nst0  IBM       ULTRIUM-HH5       G9N1**
- trying to load a tape fails (the dialog wont close
- The FILE and RESTORE tabs show an empty directory, that I can't navigate out of.
- The BACKUP tab shows the program folder. I can't navigate out it either.

Thoughts?

**Comment by samuelncui** [comment]
@patrickfitz /dev/tape/by-id/scsi-HUJ0000000 kinds of path or sg / nst path all can work. I use /dev/tape/by-id/xxx because I have multiple drives, and the sg number will change after reboot. You could use `ls -al /dev/tape/by-id/` to get a tape drive path with S/N, which will not change after reboot.

FILE and RESTORE tabs will have files when you finish your first BACKUP job, and those files in the backup job will be put into the FILE database (aks library).

The BACKUP tab shows the path that is specified in the config.yaml. If you didn't config it, as default, it will show the root path of this service.

In conclusion, you should consider editing config.yaml before using.

**Comment by swildig** [comment]
> I have a similar problem- this is not working with IBM HH LT05 drive, the IBM LTFS driver version 2.5.0.0. Installed as root. OS is openSUSE Leap 15.5.
> 
> The drive works from the command line - I can mount it from /dev/sg7 and copy files to it - but installing and following the readme didn't yield a functional system, as follows:
> 
> * it doesn't interact with the drive at all
> 
> * the default config mentions a different type of path (/dev/tape/by-id/scsi-HUJ0000000) than the recommended "sg" path
> * neither work in my case
> * the sg_map command gives me the following:  **/dev/sg7  /dev/nst0  IBM       ULTRIUM-HH5       G9N1**
> * trying to load a tape fails (the dialog wont close
> * The FILE and RESTORE tabs show an empty directory, that I can't navigate out of.
> * The BACKUP tab shows the program folder. I can't navigate out it either.
> 
> Thoughts?

If you're having an issue were the load tape dialog doesn't close check the barcode you're entering. LTO barcodes are at max 8 long and if you enter something longer the page errors and doesn't close the window. Once the window closes you'll see it start to interact with the drive.

**Comment by samuelncui** [comment]
@swildig @patrickfitz I added error popups recently, you guys can update to the newest version to check which step went wrong.

**Comment by Paruccone** [comment]
It is my understanding that the DB only gets populated at Backup time. Why if we already moved data to tapes and we want to load the list of content in YATM? Sounds like we have to restore to disk and then back up again using YATM?

Tried to run a backup and when I click on "Submit" on the tape job, nothing happens :( and nothing in logs except for first startup row:

time="2024-03-23T13:36:01Z" level=info msg="read config success, conf= '&{Domain:http://xxx:8080 Listen:xxxx:8080 DebugListen:xxx:8081 Database:{Dialect:sqlite DSN:./tapes.db} Paths:{Work:./ Source:./ Target:./} TapeDevices:[/dev/tape/by-id/scsi-HU1229P78K /dev/tape/by-id/scsi-HU1220P79Y] Scripts:{Encrypt:./scripts/encrypt Mkfs:./scripts/mkfs Mount:./scripts/mount Umount:./scripts/umount ReadInfo:./scripts/readinfo}}'"

using openltfs-2.4-stable https://github.com/LinearTapeFileSystem/ltfs/tree/v2.4-stable

If I manually load tapes, here's what I get from yatm-lto-info (seeams like it's using only one of the 2 tape drives though):
Device /dev/nst0 opened
Generic SCSI-2 tape (114)
Residual count: 0
Device registers: 58000000
Status registers: 41010000
BOT ONLINE IM_REP_EN
Error register: 0
Possibly inaccurate:
  Current file: 0
  Current block number: 0

Drive information:
   Vendor  : HP
   Model   : Ultrium 5-SCSI
   Firmware: Y6PW
Medium information:
  Cartridge Type: 0x00 - Data cartridge
  Medium format : 0x58 - LTO-5
  Formatted as  : 0x58 - LTO-5
Format specs:
   Capacity  :  1500 GB native   -  3000 GB compressed with a 2:1 ratio
   R/W Speed :   140 MB/s native -   280 MB/s compressed
   Partitions:     2 max partitions supported
   Phy. specs: 4 bands/tape, 20 wraps/band, 16 tracks/wrap, 1280 total tracks
   Duration  : 3h10 to fill tape with 80 end-to-end passes (142 seconds/pass)
Usage information:
Previous sessions:
